lib/sda/tree.go

```diff
@@ -258,7 +259,7 @@ func (tm TreeMarshal) MakeTree(il *EntityList) (*Tree, error) {
 
 // MakeTreeFromList creates a sub-tree given an EntityList
 func (tm *TreeMarshal) MakeTreeFromList(parent *TreeNode, il *EntityList) *TreeNode {
-	idx, ent := il.Search(tm.EntityId)
+	idx, ent := il.SearchEntity(tm.EntityId)
```

Comment:
So you still search the Entity-index. I would call it `EntityIdx` or something, to make it clear it's not a `TreeNodeIdx`. And put a comment in the structure saying that it's possible that one `EntityIdx` doesn't have to be unique in a tree!